lua/ibl/init.lua

```diff
@@ -365,13 +418,78 @@ M.refresh = function(bufnr)
             whitespace_tbl = fn(buffer_state.tick, bufnr, row - 1, whitespace_tbl)
         end
 
-        last_whitespace_tbl = whitespace_tbl
+        -- this do block is purely to help stylua with the scope
+        do
+            arr_whitespace_tbl[i] = whitespace_tbl
+        end
+
+        ::continue::
+    end
+
+    -- between the two loops we find the current indent column and current indent start quickly
```

Comment:
Why does this need to happen between the loop? If you keep track of the current indent in the first loop, you can just use that as the current indent when you hit the current line

Reply:
I have updated this now so it happens in the loop.